tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3973 @@
+                <td colspan="2">
+                    <details>
+                        <summary>Metadata Block</summary>
+                        <blockquote>
+                            Defines metadata (arbitrary binary data) that will be stored in the output.
+                        </blockquote>
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 2</strong> to denote a <i>Metadata Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data Length</td>
+                                <td>uint16</td>
+                                <td>
+                                    Length of the following data field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Data</td>
+                                <td>ByteArray</td>
+                                <td>Binary data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                </td>
+            </tr>
+        </table>
+    </details>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+#### Output Syntactic Validation
+
+- `Amount` field must fulfill the dust protection requirements and must not be `0`.
+- `Amount` field must be ≤ `Max IOTA Supply`.
+- `Native Tokens Count` must not be greater than `Max Native Tokens Count`.
+- `Native Tokens` must be lexicographically sorted based on `Token ID`.
+- Each <i>Native Token</i> must be unique in the set of `Native Tokens` based on its `Token ID`. No duplicates are
+  allowed.
+- `Amount` of any <i>Native Token</i> must not be `0`.
+- It must hold true that `Unlock Conditions Count` = `1`.
+- `Unlock Condition Type` of an <i>Unlock Condition</i> must define on of the following types:
+  - <i>Immutable Alias Address Unlock Condition</i>
+- It must hold true that `0` ≤ `Blocks Count` ≤ `1`.
+- `Block Type` of a <i>Block</i> in `Blocks` must define on of the following types:
+  - <i>Metadata Block</i>
+- It must hold true that `0` ≤ `Immutable Blocks Count` ≤ `1`.
+- `Block Type` of a <i>Block</i> in `Immutable Blocks` must define on of the following types:
+  - <i>Metadata Block</i>
+- Syntactic validation of all present feature blocks must pass.
+- `Token Scheme Type` must match one of the supported schemes. Any other value results in invalid output.
+- `Minted Tokens` must not be greater than `Maximum Supply`.
```

Comment:
This particular rule limits the foundry to not be able to further mint tokens once this counter hits the `MaximumSupply`. However, `MaximumSupply` implies the maximum supply that can theoretically be in circulation, meaning that that foundry should always be able to mint new tokens as long as `Minted`-`Melted` < `MaximumSupply`. This is unlike how the foundry worked previously where the `CirculatingSupply` just had to be within bounds of the `MaximumSupply` for the foundry to further mint new tokens.

Note that lifting this rule makes it so that `Minted` and `Melted` token counters can be > `MaximumSupply` but that should be fine as these counters are just indicative for the actual circulating supply. Lifting this rule also makes these counters unbound but they can never be bigger than uint256 anyway (though this is a syntactical check).

If that isn't wanted, then I suggest we rename `MaximumSupply` to `MaxMintedTokens` as to reflect this limit more clearly, however I don't know the rational of why the foundry would need to be limited in such a way.

Reply:
The first option, `Minted`-`Melted` < `MaximumSupply`, is now added to Token Scheme Validation in   https://github.com/iotaledger/tips/pull/38/commits/4257b772cab90178832ad7f43fa60d6cdd1ddc85